File: achat/schedule.py
# -*- coding: UTF-8 -*-
#
# Schedule Object
import time


STEP_NULL = -1
STEP_INIT_ENV = 0
STEP_BREAK = 1
STEP_LAST_TERM = 2
STEP_WELCOME = 3
STEP_CLASS = 4
STEP_END_TIP = 5
STEP_END = 6
STEP_CHECK = 7
STEP_ISSUE = 8

LOTTERY_PK10 = 0
LOTTERY_XYFT = 1


class Schedule(object):
    # private variables
    __step = STEP_NULL
    __lottery = LOTTERY_PK10
    __cur_time = ''
    __next_wait = 10

    # construction function
    def __init__(self):
        pass

    # private functions
    def get_short_time(self):
        min = self.__cur_time[4:5]
        sec = self.__cur_time[6:8]
        i = int(min)
        if i > 5:
            i = i - 5
        short = "{}:{}".format(i, sec)
        return short

    # public functions
    def next(self):
        self.__cur_time = time.strftime("%H:%M:%S", time.localtime())

        if self.__step == STEP_ISSUE:
            self.__step = STEP_LAST_TERM

        if self.__step == STEP_NULL:
            self.__step = self.__step + 1
            self.__next_wait = 2
            return self.__step

        if self.__cur_time >= '04:00:00' and self.__cur_time < '09:00:00':
            self.__step = STEP_BREAK
            self.__next_wait = 10
        else:
            if self.__cur_time >= '09:00:00' and self.__cur_time < '24:00:00':
                self.__lottery = LOTTERY_PK10
            else:
                self.__lottery = LOTTERY_XYFT

            short = self.get_short_time()

            if short >= '0:00' and short < '1:00':
                step = STEP_LAST_TERM
            elif short >= '1:00' and short < '1:05':
                step = STEP_WELCOME
            elif short >= '3:00' and short < '4:00':
                step = STEP_END_TIP
            elif short >= '4:00' and short < '4:05':
                step = STEP_END
            elif short >= '4:05' and short < '4:55':
                step = STEP_CHECK
            elif short >= '4:55' and short < '5:00':
                step = STEP_ISSUE
            else:
                step = STEP_CLASS

            if self.__step < step:
                self.__step = self.__step + 1

            if self.__step == STEP_LAST_TERM:
                self.__next_wait = 1
            elif self.__step == STEP_WELCOME:
                self.__next_wait = 1
            elif self.__step == STEP_END_TIP:
                self.__next_wait = 1
            elif self.__step == STEP_END:
                self.__next_wait = 2
            elif self.__step == STEP_CHECK:
                self.__next_wait = 5
            elif self.__step == STEP_ISSUE:
                self.__next_wait = 1
            else:
                self.__next_wait = 1
            pass
        return self.__step

    def get_step(self):
        return self.__step

    def get_next_wait(self):
        return self.__next_wait

    pass

File: AchatAssistant/app/src/main/java/com/hb/achat/achatassistant/Schedule.java
package com.hb.achat.achatassistant;


import android.util.Log;

public class Schedule {
    public static final int STEP_NULL = -1;
    public static final int STEP_INIT_ENV = 0;
    public static final int STEP_BREAK = 1;
    public static final int STEP_LAST_TERM = 2;
    public static final int STEP_WELCOME = 3;
    public static final int STEP_CLASS = 4;
    public static final int STEP_END_TIP = 5;
    public static final int STEP_END = 6;
    public static final int STEP_CHECK = 7;
    public static final int STEP_ISSUE = 8;

    public static final int LOTTERY_PK10 = 0;
    public static final int LOTTERY_XYFT = 1;

    private int mStep = STEP_NULL;
    private int mLottery = LOTTERY_PK10;
    private String mCurTime = "";
    private int mNextWait = 10;

    private String getShortTime() {
        String min = mCurTime.substring(4, 5);
        String sec = mCurTime.substring(6, 8);
        int i = Integer.parseInt(min);
        if (i > 5) {
            i = i - 5;
        }
        return String.format("%d:%s", i, sec);
    }

    public int next() {
        mCurTime = Tools.getCurTimeFormatted();
        //Log.d("AASERVICE", "Schedule, current time: " + mCurTime);
        if (mStep == STEP_ISSUE) {
            mStep = STEP_LAST_TERM;
        }

        if (mStep == STEP_NULL) {
            mStep = mStep + 1;
            mNextWait = 2;
            return mStep;
        }

        if (mCurTime.compareTo("04:00:00") >= 0 && mCurTime.compareTo("09:00:00") < 0) {
            mStep = STEP_BREAK;
            mNextWait = 10;
        } else {
            String shortTime = getShortTime();
            //Log.d("AASERVICE", "Schedule, short time: " + shortTime);
            int step = 0;

            if (mCurTime.compareTo("00:05:00") >= 0 && mCurTime.compareTo("04:05:00") < 0) {
                mLottery = LOTTERY_XYFT;
                if (shortTime.compareTo("0:00") >= 0 && shortTime.compareTo("0:30") < 0) {
                    step = STEP_LAST_TERM;
                } else if (shortTime.compareTo("0:30") >= 0 && shortTime.compareTo("0:40") < 0) {
                    step = STEP_WELCOME;
                } else if (shortTime.compareTo("0:40") >= 0 && shortTime.compareTo("3:00") < 0) {
                    step = STEP_CLASS;
                } else if (shortTime.compareTo("3:00") >= 0 && shortTime.compareTo("4:00") < 0) {
                    step = STEP_END_TIP;
                } else if (shortTime.compareTo("4:00") >= 0 && shortTime.compareTo("4:05") < 0) {
                    step = STEP_END;
                } else if (shortTime.compareTo("4:05") >= 0 && shortTime.compareTo("4:55") < 0) {
                    step = STEP_CHECK;
                } else if (shortTime.compareTo("4:55") >= 0 && shortTime.compareTo("5:00") < 0) {
                    step = STEP_ISSUE;
                } else {
                    step = STEP_CLASS;
                }
            } else {
                mLottery = LOTTERY_PK10;
                if (shortTime.compareTo("3:00") >= 0 && shortTime.compareTo("3:30") < 0) {
                    step = STEP_LAST_TERM;
                } else if (shortTime.compareTo("3:30") >= 0 && shortTime.compareTo("3:40") < 0) {
                    step = STEP_WELCOME;
                } else if (shortTime.compareTo("3:40") >= 0) {
                    step = STEP_CLASS;
                } else if (shortTime.compareTo("1:00") >= 0 && shortTime.compareTo("2:00") < 0) {
                    step = STEP_END_TIP;
                } else if (shortTime.compareTo("2:00") >= 0 && shortTime.compareTo("2:05") < 0) {
                    step = STEP_END;
                } else if (shortTime.compareTo("2:05") >= 0 && shortTime.compareTo("2:55") < 0) {
                    step = STEP_CHECK;
                } else if (shortTime.compareTo("2:55") >= 0 && shortTime.compareTo("3:00") < 0) {
                    step = STEP_ISSUE;
                } else {
                    step = STEP_CLASS;
                }
            }

            if (mStep < step) {
                mStep = mStep + 1;
            }

            switch (mStep) {
                case STEP_LAST_TERM:
                    mNextWait = 1;
                    break;
                case STEP_WELCOME:
                    mNextWait = 1;
                    break;
                case STEP_END_TIP:
                    mNextWait = 1;
                    break;
                case STEP_END:
                    mNextWait = 2;
                    break;
                case STEP_CHECK:
                    mNextWait = 2;
                    break;
                case STEP_ISSUE:
                    mNextWait = 5;
                    break;
                default:
                    mNextWait = 1;
                    break;
            }
        }
        return mStep;
    }

    public int getStep() {
        return mStep;
    }

    public int getNextWait() {
        return mNextWait;
    }

}
